clcache.py

```diff
@@ -1384,10 +1384,7 @@ def processDirect(cache, outputFile, compiler, cmdLine, sourceFile):
             listOfHeaderHashes = []
             for fileName in manifest.includeFiles:
                 fileHash = getFileHash(expandBasedirPlaceholder(fileName, baseDir))
-                if fileHash is not None:
-                    # May be if source does not use this header anymore (e.g. if that
-                    # header was included through some other header, which now changed).
-                    listOfHeaderHashes.append(fileHash)
+                listOfHeaderHashes.append(fileHash)
```

Comment:
I just realized the issue is much more global. A missing include can have multiple meanings:

1. The file was deleted for no reason -> we must not restore the object because it is a compiler error
2. One include changed, making another obsolete. In this case the entire manifest is invalid as it is defined by the list of includes.

I think we need a way to get into the `postprocessHeaderChangedMiss` branch for both cases. Somethink like:

```python
            try:
                includesContentHash = ManifestsManager.getIncludesContentHashForFiles(
                    [expandBasedirPlaceholder(include, baseDir) for include in manifest.includeFiles])
                cachekey = manifest.includesContentToObjectMap.get(includesContentHash)
                if cachekey is None:
                    raise HeaderChanged()

                if cache.hasEntry(cachekey):
                    return processCacheHit(cache, objectFile, cachekey)
                else:
                    postProcessing = lambda compilerResult: postprocessObjectEvicted(
                        cache, objectFile, cachekey, compilerResult)
            except HeaderChanged e:
                postProcessing = lambda compilerResult: postprocessHeaderChangedMiss(
                    cache, objectFile, manifest, manifestHash, includesContentHash, compilerResult)
```